lib/puppetserver/ca/action/delete.rb

```diff
@@ -107,7 +106,70 @@ def run(args)
           # Perform the desired action, keeping track if any errors occurred
           errored = false
           deleted_count = 0
-          inventory_file_path = File.join(settings[:cadir], 'inventory.txt')
+          cadir = settings[:cadir]
+          inventory_file_path = File.join(cadir, 'inventory.txt')
+
+          # Because --revoke has a potentially fatal error it can throw,
+          # process it first.
+          if args['revoked']
+            loader = X509Loader.new(settings[:cacert], settings[:cakey], settings[:cacrl])
+            verified_crls = loader.crls.select { |crl| crl.verify(loader.key) }
+            unless verified_crls.length == 1
+              @logger.err("Could not identify Puppet's CRL. Aborting delete action.")
+              return 1
+            end
+            crl = verified_crls.first
+
+            # First, search the inventory for the given serial. 
```

Comment:
Isn't it 'revoked serial' instead of 'given serial'?

Reply:
Yeah, suppose it is :)